ethers-solc/src/compile/output/mod.rs

```diff
@@ -426,8 +428,14 @@ impl AggregatedCompilerOutput {
     }
 
     /// Whether the output contains a compiler error
-    pub fn has_error(&self) -> bool {
-        self.errors.iter().any(|err| err.severity.is_error())
+    pub fn has_error(&self, warnings_as_errors: bool) -> bool {
```

Comment:
I think this is fine.

Instead of the bool argument this could just take the `Severity` enum as argument and find all errors that are `>=` (needs Ord impl)

Reply:
I pushed a second commit that changed to using Severity as argument. I'm assuming it was following the correct pattern. If you think there are any specifics that could be adjusted, feel free to let me know.

During implement I had to choose between adding a Copy trait to Severity or passing a pointer down. I chose pointer since it was similar to "ignored_error_codes" and wasn't sure if Copy is something that is generally not preferred?